### PR TITLE
[tests-only] created resources rather than using skeleton dirs with users in webUI suites

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
created resources rather than using skeleton dirs with users in webUI suites

## Related Issue

- Part of https://github.com/owncloud/QA/issues/658

## How Has This Been Tested?

- 🤖 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/webUICreateDelete/deleteFilesFolders.feature [hunk 1]
```diff
@@ -71,6 +75,7 @@ Feature: deleting files and folders
   @smokeTest @skipOnLDAP
   @skipOnEncryption @encryption-issue-74
   Scenario: Delete multiple files at once
+
```
Comment (saw-jan) [inline tests/acceptance/features/webUICreateDelete/deleteFilesFolders.feature:78]: extra line?

File: tests/acceptance/features/webUIFavorites/favoritesFile.feature [hunk 2]
```diff
@@ -37,7 +37,14 @@ Feature: Mark file as favorite
     And folder "simple-empty-folder" should not be listed in the favorites page on the webUI
 
   Scenario: mark files with same name and different path as favorites and list them in favourites page
-    Given user "Alice" has been created with default attributes and large skeleton files
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And user "Alice" has created folder "simple-empty-folder"
+    And user "Alice" has created folder "simple-folder"
+    And user "Alice" has created folder "simple-folder/simple-empty-folder"
+    And user "Alice" has created folder "strängé नेपाली folder"
```
Comment (saw-jan) [inline tests/acceptance/features/webUIFavorites/favoritesFile.feature:44]: you can make use of this step
```
Given user "Alice" has created the following folders
      | path         |
```

Comment (kiranparajuli589) [reply tests/acceptance/features/webUIFavorites/favoritesFile.feature]: nice.... I've used that! have a look again!

**Review by Talank — COMMENTED** [review]
LGTM

File: tests/acceptance/features/webUIFavorites/favoritesFile.feature [hunk 3]
```diff
@@ -37,7 +40,15 @@ Feature: Mark file as favorite
     And folder "simple-empty-folder" should not be listed in the favorites page on the webUI
 
   Scenario: mark files with same name and different path as favorites and list them in favourites page
-    Given user "Alice" has been created with default attributes and large skeleton files
+    Given user "Alice" has created the following folders
+      | path                              |
+      | simple-folder                     |
+      | simple-empty-folder               |
+      | strängé नेपाली folder             |
```
Comment (saw-jan) [inline tests/acceptance/features/webUIFavorites/favoritesFile.feature:47]: minor indentation to fix

File: tests/acceptance/features/webUIFiles/search.feature [hunk 4]
```diff
@@ -6,12 +6,24 @@ Feature: Search
...
-    And the user has browsed to the files page
+    Given user "Alice" has been created with default attributes and without skeleton files
 
   @smokeTest
   Scenario: Simple search
+    Given user "Alice" has created the following folders
+      | path                  |
+      | 0                     |
+      | simple-folder         |
+      | strängé नेपाली folder |
```
Comment (saw-jan) [inline tests/acceptance/features/webUIFiles/search.feature:17]: minor indentation to fix

File: tests/acceptance/features/webUIFiles/search.feature [hunk 5]
```diff
@@ -22,6 +34,16 @@ Feature: Search
     And file "lorem-big.txt" with path "/strängé नेपाली folder" should be listed in the search results in the other folders section on the webUI
 
   Scenario: search for folders
+    Given user "Alice" has created the following folders
+      | path                  |
+      | simple-folder         |
+      | 'single'quotes        |
+      | strängé नेपाली folder |
```
Comment (saw-jan) [inline tests/acceptance/features/webUIFiles/search.feature:41]: minor indentation to fix

File: tests/acceptance/features/webUIFiles/search.feature [hunk 6]
```diff
@@ -33,6 +55,20 @@ Feature: Search
     And file "lorem.txt" should not be listed in the search results in the other folders section on the webUI
 
   Scenario: search in sub folder
+    Given user "Alice" has created the following folders
+      | path                  |
+      | 0                     |
+      | simple-folder         |
+      | strängé नेपाली folder |
```
Comment (saw-jan) [inline tests/acceptance/features/webUIFiles/search.feature:62]: minor indentation to fix

File: tests/acceptance/features/webUIFavorites/favoritesFile.feature [hunk 7]
```diff
@@ -37,7 +40,15 @@ Feature: Mark file as favorite
     And folder "simple-empty-folder" should not be listed in the favorites page on the webUI
 
   Scenario: mark files with same name and different path as favorites and list them in favourites page
-    Given user "Alice" has been created with default attributes and large skeleton files
+    Given user "Alice" has created the following folders
+      | path                              |
+      | simple-folder                     |
+      | simple-empty-folder               |
+      | strängé नेपाली folder                |
+      | simple-folder/simple-empty-folder |
```
Comment (swoichha) [inline tests/acceptance/features/webUIFavorites/favoritesFile.feature:48]: Indentation

File: tests/acceptance/features/webUIFiles/search.feature [hunk 8]
```diff
@@ -6,12 +6,24 @@ Feature: Search
...
-    And the user has browsed to the files page
+    Given user "Alice" has been created with default attributes and without skeleton files
 
   @smokeTest
   Scenario: Simple search
+    Given user "Alice" has created the following folders
+      | path                  |
+      | 0                     |
+      | simple-folder         |
+      | strängé नेपाली folder |  |
```
Comment (swoichha) [inline tests/acceptance/features/webUIFiles/search.feature]:
```suggestion
      | strängé नेपाली folder    | 
```

File: tests/acceptance/features/webUIFiles/search.feature [hunk 9]
```diff
@@ -22,6 +34,16 @@ Feature: Search
     And file "lorem-big.txt" with path "/strängé नेपाली folder" should be listed in the search results in the other folders section on the webUI
 
   Scenario: search for folders
+    Given user "Alice" has created the following folders
+      | path                  |
+      | simple-folder         |
+      | 'single'quotes        |
+      | strängé नेपाली folder |  |
```
Comment (swoichha) [inline tests/acceptance/features/webUIFiles/search.feature]:
```suggestion
      | strängé नेपाली folder    | 
```

File: tests/acceptance/features/webUIFiles/search.feature [hunk 10]
```diff
@@ -100,37 +154,52 @@ Feature: Search
...
   Scenario: Search for a shared folder
-    Given user "Carol" has been created with default attributes and without skeleton files
+    Given user "Alice" has created folder "simple-folder"
+    And user "Alice" has logged in using the webUI
+    And the user has browsed to the files page
+    And user "Carol" has been created with default attributes and without skeleton files
     And user "Carol" has created folder "simple-folder"
-    When user "Carol" shares folder "simple-folder" with user "Alice" using the sharing API
+    And user "Carol" has shared folder "simple-folder" with user "Alice"
     And the user reloads the current page of the webUI
```
Comment (swoichha) [inline tests/acceptance/features/webUIFiles/search.feature:163]:
```suggestion
    When the user reloads the current page of the webUI
```

**Review by swoichha — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/webUIFiles/search.feature (hunk 10 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUIFiles/search.feature:163]: its converted to given step. the scenario does not need sharing as a when step

Comment (saw-jan) [reply tests/acceptance/features/webUIFiles/search.feature:163]: she means this step `And the user reloads the current page of the webUI`

Comment (swoichha) [reply tests/acceptance/features/webUIFiles/search.feature:163]: Yes, @saw-jan is right. IMO the `And the user reloads the current page of the webUI` is the `When` step.

Comment (kiranparajuli589) [reply tests/acceptance/features/webUIFiles/search.feature:163]: thank you for pointing. but it can go as a given step too..
fixed!

**Review by saw-jan — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/webUIFiles/search.feature [hunk 11]
```diff
@@ -22,6 +34,17 @@ Feature: Search
...
   Scenario: search for folders
+    Given user "Alice" has created the following folders
+      | path                               |
+      | simple-folder                      |
+      | 'single'quotes                     |
+      | 'single'quotes/simple-empty-folder |
+      | strängé नेपाली folder                |
+    And user "Alice" has uploaded file "filesForUpload/zzzz-must-be-last-file-in-folder.txt" to "zzzz-must-be-last-file-in-folder.txt"
+    And user "Alice" has uploaded file "filesForUpload/zzzz-must-be-last-file-in-folder.txt" to "simple-folder/zzzz-must-be-last-file-in-folder.txt"
+    And user "Alice" has uploaded file "filesForUpload/zzzz-must-be-last-file-in-folder.txt" to "strängé नेपाली folder/zzzz-must-be-last-file-in-folder.txt"
```
Comment (individual-it) [inline tests/acceptance/features/webUIFiles/search.feature:45]: please create also a `lorem.txt` file to make sure the search algorithm works correctly. We check that lorem.txt is not viewed in the results, but if we don't have that file in the first place this test becomes irrelevant.
Please also double check other tests for the same issue

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/webUIFiles/search.feature (hunk 11 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUIFiles/search.feature:45]: fixed!

File: tests/acceptance/features/webUIFiles/search.feature [hunk 12]
```diff
@@ -100,37 +157,52 @@ Feature: Search
...
   Scenario: Search for files with difficult names
-    Given user "Alice" has uploaded file with content "does-not-matter" to "/strängéनेपालीloremfile.txt"
+    Given user "Alice" has created folder "strängé नेपाली folder"
+    And user "Alice" has uploaded file with content "does-not-matter" to "/strängéनेपालीloremfile.txt"
     And user "Alice" has uploaded file with content "does-not-matter" to "/strängé नेपाली folder/strängéनेपालीloremfile.txt"
+    And user "Alice" has logged in using the webUI
+    And the user has browsed to the files page
     When the user searches for "lorem" using the webUI
-    Then file "strängéनेपालीloremfile.txt" with path "/" should be listed in the search results in the other folders section on the webUI
+    Then file "strängéनेपालीloremfile.txt" should be listed on the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUIFiles/search.feature:197]: why this change?

Comment (kiranparajuli589) [reply tests/acceptance/features/webUIFiles/search.feature:197]: I was also confused by this. If I run the test with a large skeleton, the then step succeeds, but if I created the files, its failing
@individual-it do you see something missing here??

**Review by swoichha — APPROVED** [review]
LGTM

File: tests/acceptance/features/webUIFiles/search.feature (hunk 12 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUIFiles/search.feature:200]: I've created an issue for this https://github.com/owncloud/core/issues/38641
Actually this test scenario should have reload after the files upload...so I've fixed the scenario. The scenario above `Search for a newly uploaded file` this demonstrates the bug issued.

**Review by individual-it — APPROVED** [review]

**Pull request merged**